src/moi.jl

```diff
@@ -428,8 +475,32 @@ function build_bilevel(
                 # feasible equality constraints always satisfy complementarity
             end
         end
+
+        if consider_constrained_variables
+            # complementary slackness for variable bounds
+            variable_comps = get_variable_complements(lower, lower_dual, lower_primal_dual_map)
+            for comp in variable_comps
+                if !is_equality(comp.set_w_zero)
+                    accept_vector_set(mode, comp)
+                    add_complement(
+                        mode,
+                        m,
+                        comp,
+                        lower_dual_idxmap,
+                        lower_idxmap,
+                        copy_names,
+                        pass_start,
+                    )
+                end
+            end
+        end
+
     else # strong duality
-        add_strong_duality(mode, m, lower_primal_obj, lower_dual_obj, lower_idxmap, lower_dual_idxmap)
+        if !consider_constrained_variables || contains_only_scalar_sets(lower)
+            add_strong_duality(mode, m, lower_primal_obj, lower_dual_obj, lower_idxmap, lower_dual_idxmap)
+        else
+            error("Only scalar sets in lower level allowed with consider_constrained_variables.")
+        end
```

Comment:
I dont think any of this is needed

Reply:
true, this threw occasional errors earlier. Found out that this was related to missing dual variables for which start values could not be passed. 
This is now resolved, since hints etc. are no longer passed if the dual to bounds does not exist in the first place...